subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -1085,7 +1085,7 @@ private boolean hasSettingsFile(TestFile dir) {
         if (useOwnUserHomeServices || useCustomGradleUserHomeDir) {
             properties.put(REUSE_USER_HOME_SERVICES, "false");
         }
-        if (!noExplicitTmpDir) {
+        if (!noExplicitTmpDir && buildJvmOpts.stream().noneMatch(arg -> arg.startsWith("-Djava.io.tmpdir="))) {
```

Comment:
Why do we need `noExplicitTmpDir` any more? All the tests which use `noExplictTmpDir` also set `java.io.tmpdir` now, right?


I also wonder what those tests which used `withNoExplicitTmpDir()` have been testing. Since it seems like they are not testing the same thing any more, right?

Reply:
Only 1 test is using `withNoExplicitTmpDir`, and it's very old. Let's remove it to see if anything breaks.